### DISCUSSION TITLE
Question about the MS C++ redist

### DISCUSSION THREAD

**Discussion by NikAWing** [TumblThreeApp/TumblThree#364] [Q&A (Question & Answer)]
Hello :)

I'm trying to find out if the new releases work for me again (0 byte file problem).
I was sure that I already installed the VC++ redist in 2015 but the program crashes while starting up and so I followed the link to the redist file. Trying to install it results in an error, the file is older than the one I have installed (14.0.23026 vs 14.0.24210)
I also have the VC++ redist 2015-2019 x64 (14.26.28720) installed.

What can I do (except uninstalling the newer version(s) which usually is not recommended)?

Best regards!

**Comment by thomas694** [comment]
Normally that isn't a problem, but you never know.
I assume you use the x64 version of TumblThree, so you need the VC++ redist x64 version which you already use in a newer version.
You could install the newest "VC++ redist 2015-2019" [version (14.28.29913)](https://support.microsoft.com/en-us/topic/the-latest-supported-visual-c-downloads-2647da03-1eea-4433-9aff-95f26a218cc0) for [x86](https://aka.ms/vs/16/release/vc_redist.x86.exe) and [x64](https://aka.ms/vs/16/release/vc_redist.x64.exe), and see if it helps.

But it would be also good, if you have a look in our log file (TumblThree.log) and the one of the chromium component (debug.log). You don't need to post the whole content, only the error right before the crash, if there's a log entry.

**Comment by NikAWing** [comment]
Updating to the newest 2015-2019 did not change anything.

I tried to generate the log file like described here: https://github.com/TumblThreeApp/TumblThree/issues/21#issuecomment-699618439
but nothing is created anywhere. I don't see TumblThree.log or debug.log in any folder (neither in program folder, appdata folder, temp, etc)

If I can't fix it I will try replacing the newer 2015 redist with the older one.

**Comment by thomas694** [comment]
The step described in issue 21 is no longer necessary, it can be configured in the settings dialog now.
The log file will be written to the folder %LOCALAPPDATA%\TumblThree or the current folder (portable mode).

**Comment by NikAWing** [comment]
ok, how do I get to the settings dialog?
I suppose I have to set it in the newest version to make this work?
I tried setting protocol level to verbose in 1.0.11.13 and no log files appear for neither this or newest version (from starting and closing - or crashing)

**Comment by thomas694** [comment]
The Settings button is at the bottom of the window, there it's inside the General tab.
Logging in that way should work since version 1.0.12.1.
But some users report that to them, for some unknown reason, it's not working at all.

**Comment by NikAWing** [comment]
just tested the most recent version where you changed the vcredist to 2015-2019 :)
but still, the usual crash happens.

it's a System.NullReferenceException (and also was when I opened up this issue about the previous version 1.3.0)

I tried setting portable mode to have logs in the program directory, but no logs appear.

I tried opening the project source to see where it comes from, but I have VS2013 here and get some errors.
If I can be of any help, let me know :)

**Comment by thomas694** [comment]
Yes, VS2013 cannot be used, because we use some language elements from newer version, so minimum would be VS2015.
Anyway, [VS2019 Community](https://visualstudio.microsoft.com/vs/community/) (setup option/workload ".NET desktop development" should be sufficient) is free for open source and individuals and can be installed side-by-side to older versions.

Sure, it would help us, if you can find out where it crashes and copy the exception information.

**Comment by NikAWing** [comment]
:thinking: I usually don't like to do something like that, but I deleted the files in appdata and the program loads again ...
:shrug:

**Comment by thomas694** [comment]
I'm glad it works for you again. It would have been interesting what was the problem with your TumblThree profile folder. But if you haven't made a backup of the folder, we can hardly find out.

**Comment by thomas694** [comment]
The issue has been closed. You can still comment. Feel free to ask for reopening the issue if needed.
